### PR TITLE
docs: update wording of sync.md

### PR DESCRIPTION
**Pull request by CaitinChen**

### CODE REVIEW

File: docs/en/guide/sync.md [hunk 1]
```diff
@@ -1,355 +1,265 @@
 ---
 title: Data Synchronization
-sidebar_position: 7
-description: Learn how to use the data sync tool in JuiceFS.
+sidebar_position: 9
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:3]:
```suggestion
sidebar_position: 7
```

File: docs/en/guide/sync.md [hunk 2]
```diff
@@ -1,355 +1,265 @@
...
+description: Learn how to use juicefs sync for efficient data synchronization across supported storage systems, including object storage, JuiceFS, and local file systems.
 ---
 
-[`juicefs sync`](../reference/command_reference.mdx#sync) is a powerful data migration tool, which can copy data across all supported storages including object storage, JuiceFS itself, and local file systems, you can freely copy data between any of these systems. In addition, it supports remote directories through SSH, HDFS, WebDAV, etc. while providing advanced features such as  incremental synchronization, and pattern matching (like rsync), and distributed syncing.
+[`juicefs sync`](../reference/command_reference.md#sync) is a powerful data synchronization tool that can copy data across all supported storage systems, including object storage, JuiceFS, and local file systems. You can freely copy data between any of these systems. It also supports syncing remote directories accessed via SSH, HDFS, and WebDAV. Advanced features include incremental synchronization, pattern matching (like rsync), and distributed syncing.
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:7]:
```suggestion
[`juicefs sync`](../reference/command_reference.mdx#sync) is a powerful data synchronization tool that can copy data across all supported storage systems, including object storage, JuiceFS, and local file systems. You can freely copy data between any of these systems. It also supports syncing remote directories accessed via SSH, HDFS, and WebDAV. Advanced features include incremental synchronization, pattern matching (like rsync), and distributed syncing.
```

File: docs/en/guide/sync.md [hunk 3]
```diff
@@ -1,355 +1,265 @@
...
-
-If an element matches a certain pattern, there will be two processing logics:
+By default, `juicefs sync` performs incremental synchronization. It only overwrites files if their sizes are different. You can also use [`--update`](../reference/command_reference.md#sync) to overwrite files when the `mtime` of the source file has been updated. For scenarios with higher demand for data integrity, use [`--check-new`](../reference/command_reference.md#sync) or [`--check-all`](../reference/command_reference.md#sync) to perform byte-by-byte comparison between the source and the destination.
 
-- If the pattern is an exclude pattern, the "exclude" behavior will be returned directly as the final matching result;
-- If the pattern is an include pattern, the subsequent patterns to be matched at this layer are skipped and the next layer is entered directly.
+For full synchronization (where all files are synchronized regardless of their presence on the destination path), use [`--force-update`](../reference/command_reference.md#sync).
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:148]: @timfeirg PTAL

File: docs/en/guide/sync.md [hunk 4]
```diff
@@ -1,355 +1,265 @@
...
 
-The behavior of layer-by-layer filtering mode is more complicated to understand and use, but it is basically compatible with rsync's `--include/--exclude` options, so it is generally recommended to be used in scenarios that require compatibility with rsync behavior.
+* When the destination is a JuiceFS file system, use the `jfs://` protocol, because it bypasses the FUSE mount point (reducing overhead) and is already optimized for file fragmentation problems. See the next point for details.
 
-### Directory structure and file permissions {#directory-structure-and-file-permissions}
+* If you must use a FUSE mount point for data synchronization, for large file transfers, it is recommended to add the [`--flush-wait=60s`](../reference/command_reference.md#object-storage-options) option. This effectively helps reduce the number of slices and mitigates write amplification. For more information, see [troubleshooting write amplification](https://github.com/juicedata/docs/pull/administration/troubleshooting.md#write-amplification).
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:175]: @timfeirg PTAL

File: docs/en/guide/sync.md [hunk 5]
```diff
@@ -1,355 +1,265 @@
...
-```shell
-# mount JuiceFS
-juicefs mount -d redis://10.10.0.8:6379/1 /mnt/jfs
-# synchronization
-juicefs sync /mnt/jfs/ s3://ABCDEFG:HIJKLMN@aaa.s3.us-west-1.amazonaws.com/
-```
+If you notice the progress bar is not changing, use the methods below for monitoring and troubleshooting:
 
-After sync, you can see all the files in [Object Storage A](#required-storages).
+* Add the [`--verbose` or `--debug`](../reference/command_reference.md#global-options) option to the `juicefs sync` command to print debug logs.
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:233]: @timfeirg PTAL

File: docs/en/guide/sync.md [hunk 6]
```diff
@@ -1,355 +1,265 @@
...
 
 ![sync via public metadata service](../images/sync-public-metadata.svg)
 
-S3 Gateway comes to rescue in these circumstances: deploy a gateway in the source region, and since this gateway accesses metadata via private network, metadata latency is eliminated to a minimum, bringing the best performance for small file intensive scenarios.
+JuiceFS S3 Gateway is the solution in these scenarios: by deploying a gateway in the source region, metadata is accessed over a private network, minimizing metadata latency and delivering optimal performance for small-file-intensive workloads.
 
 ![sync via gateway](../images/sync-via-gateway.svg)
 
-Read [S3 Gateway](../deployment/s3_gateway.md) to learn its deployment and use.
+Read [S3 Gateway](../s3_gateway.md) to learn its deployment and use.
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:265]:
```suggestion
Read [JuiceFS S3 Gateway](../deployment/s3_gateway.md) to learn its deployment and use.
```

File: docs/en/guide/sync.md [hunk 7]
```diff
@@ -1,355 +1,265 @@
...
 
-1. **Object Storage A**
-   - Bucket name: aaa
-   - Endpoint: `https://aaa.s3.us-west-1.amazonaws.com`
+The `sync` command supports two filtering modes: *full path filtering* and *layer-by-layer filtering*. Both use `--include` and `--exclude` to filter files, but their behaviors are different. By default, `sync` employs the layer-by-layer filtering mode, which is more complicated but resembles rsync's usage. Therefore, it is only recommended for users familiar with rsync. For most people, `--match-full-path` is recommended because it is much easier to understand.
 
-2. **Object Storage B**
-   - Bucket name: bbb
-   - Endpoint: `https://bbb.oss-cn-hangzhou.aliyuncs.com`
+### Full path filtering (recommended) {#full-path-filtering-mode}
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:70]:
```suggestion
### Full path filtering (recommended) <VersionAdd>1.2.0</VersionAdd> {#full-path-filtering-mode}
```

File: docs/en/guide/sync.md [hunk 8]
```diff
@@ -1,355 +1,265 @@
...
 sidebar_position: 7
-description: Learn how to use the data sync tool in JuiceFS.
+description: Learn how to use juicefs sync for efficient data synchronization across supported storage systems, including object storage, JuiceFS, and local file systems.
 ---
 
-[`juicefs sync`](../reference/command_reference.mdx#sync) is a powerful data migration tool, which can copy data across all supported storages including object storage, JuiceFS itself, and local file systems, you can freely copy data between any of these systems. In addition, it supports remote directories through SSH, HDFS, WebDAV, etc. while providing advanced features such as  incremental synchronization, and pattern matching (like rsync), and distributed syncing.
+[`juicefs sync`](../reference/command_reference.mdxx#sync) is a powerful data synchronization tool that can copy data across all supported storage systems, including object storage, JuiceFS, and local file systems. You can freely copy data between any of these systems. It also supports syncing remote directories accessed via SSH, HDFS, and WebDAV. Advanced features include incremental synchronization, pattern matching (like rsync), and distributed syncing.
```
Comment (CaitinChen) [inline docs/en/guide/sync.md:7]:
```suggestion
[`juicefs sync`](../reference/command_reference.mdx#sync) is a powerful data synchronization tool that can copy data across all supported storage systems, including object storage, JuiceFS, and local file systems. You can freely copy data between any of these systems. It also supports syncing remote directories accessed via SSH, HDFS, and WebDAV. Advanced features include incremental synchronization, pattern matching (like rsync), and distributed syncing.
```

**Comment by timfeirg** [comment]
I'll take over this pr and make changes to TOC to align most content with EE docs https://juicefs.com/docs/zh/cloud/guide/sync/

**Pull request merged**